### PR TITLE
Adding port mapping for new Arista Hardware SKUs

### PR DESCRIPTION
**Pull request by arista-hpandya**
### Description of PR

Primiarly add support for various Arista 7800R3A SKUs s to port_utils.py. Also one minor tweak `config_sonic_basedon_testbed.yml` to parameterize `asic_topo_config` per slot.

Summary:

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [x] 202405

#### What is the motivation for this PR? How did you do it?
Support all 7800R3A SKU variants

#### How did you verify/test it?
Validated with sonic-mgmt runs against Arista 7800R3A linecards.

#### Any platform specific information?
NA

#### Supported testbed topology if it's a new test case?
NA

### Documentation

This is not a new feature and hence does not require a Wiki change.

### CODE REVIEW

**Comment by wsycqyz** [comment]
The pre-commit check is a mandatory. Please help fix it.

**Comment by kenneth-arista** [comment]
@arlakshm fyi

File: ansible/config_sonic_basedon_testbed.yml [hunk 1]
```diff
@@ -279,7 +279,7 @@
   - block:
       - name: Create new dictionary for my slot
         set_fact:
-          new_asic_topo_config : "{{ new_asic_topo_config | default( { slot_num  : asic_topo_config.slot0 }) }}"
+          new_asic_topo_config : "{{ new_asic_topo_config | default( { slot_num  : asic_topo_config[slot_num] }) }}"
```
Comment (arlakshm) [inline ansible/config_sonic_basedon_testbed.yml:282]: why do we need this change?

Comment (arista-hpandya) [reply ansible/config_sonic_basedon_testbed.yml]: Hi @arlakshm, I have removed this change from the current PR. I will talk with the team and if the template change is needed, I will go ahead and create a separate PR.

Comment (arista-hpandya) [reply ansible/config_sonic_basedon_testbed.yml]: An update on this, I had a conversation with Song and we have concluded that the config change is not needed.

**Comment by arista-hpandya** [comment]
@arlakshm I have addressed the issue with the config change, should we go ahead with the merge? If there are any other concerns/suggestions blocking the merge, let me know and we can discuss them.

**Review by arlakshm — APPROVED** [review]

**Comment by vmittal-msft** [comment]
@arista-hpandya @kenneth-arista  we don't need this in 202205 ?

**Comment by kenneth-arista** [comment]
> @arista-hpandya @kenneth-arista we don't need this in 202205 ?

This is not critical for 202205 since this release is mostly frozen and not accepting test infra changes. Internally, we have this change patched. We would like to merge this change to `master` to avoid having to carry it internally.

**Comment by wenyiz2021** [comment]
close and reopen to trigger semgrep

**Pull request merged**